### PR TITLE
Season stats

### PR DESCRIPTION
**Pull request by andrew-bingham1**
- Complete season_stats methods w/ refactors and tests

### CODE REVIEW

File: lib/game_teams.rb [hunk 1]
```diff
@@ -16,7 +16,7 @@ def initialize(details)
     @tackles = details[:tackles]
     @team_id = details[:team_id]
     @game_id = details[:game_id]
-    @season_id = @game_id[0..3]
+    @season_id = "#{@game_id[0..3].to_i}#{@game_id[0..3].to_i.next}"
```
Comment (thawley2) [inline lib/game_teams.rb:19]: I will say it again.  Genius!!!!

**Review by angelbyun — APPROVED** [review]
Methods and testing look great. May need to refactor some of the methods with helper methods and add some testing for the helper methods.

File: lib/game_teams.rb (hunk 1 above)
Comment (CrowRising) [reply lib/game_teams.rb:19]: agreed!!!!

File: lib/season_stats.rb [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    coaches = game_total(season_year).merge(coach_win(season_year)) {|coach, games, wins| wins.to_f / games}
+    coaches.key(coaches.values.max)
+  end
+
+  def worst_coach(season_year)
+    coaches = game_total(season_year).merge(coach_win(season_year)) {|coach, games, wins| wins.to_f / games}
+    coaches.key(coaches.values.min)
+  end
+
+  def tackles_total(season)
```
Comment (thawley2) [inline lib/season_stats.rb:93]: Great refactor

File: lib/season_stats.rb [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        @coach_wins[game.coach] = 0 
+      end
+    end
+    total_game_hash
+  end
+
+  def coach_win(season_year)
+    @game_teams.each do |game|
+      if game.season_id == season_year 
+         if game.result == "WIN"
```
Comment (thawley2) [inline lib/season_stats.rb:75]: We could do a && instead of two if statements

**Review by thawley2 — APPROVED** [review]
Season stats Complete.... for now.

**Review by CrowRising — APPROVED** [review]
Really cool to see the methods y'all created and to integrate the data sets from the other csv files that we didn't use in ours, looking forward to the stat tracker portion   of the progress to see al of our work come together

**Pull request merged**